src/ptf/testutils.py

```diff
@@ -1406,4 +1536,72 @@ def verify_packets_any(test, pkt, ports=[], device_number=0):
 
     test.assertTrue(received == True, "Did not receive pkt on any of ports %r for device %d" % (ports, device_number))
 
+def verify_any_packet_any_port(test, pkts=[], ports=[], device_number=0):
+    """
+    Check that _any_ of the packet is received on _any_ of the specified ports belonging to
+    the given device (default device_number is 0).
+
+    Also verifies that the packet is ot received on any other ports for this
+    device, and that no other packets are received on the device (unless --relax
+    is in effect).
+
+    Returns the index of the port on which the packet is recevied.
+    """
+    received = False
+    match_index = 0
+    (rcv_device, rcv_port, rcv_pkt, pkt_time) = dp_poll(
+        test,
+        device_number=device_number,
+        timeout=1
+    )
+
+    logging.debug("Checking for pkt on device %d, port %r", device_number, ports)
+    if rcv_port in ports:
+        for pkt in pkts:
+            if str(pkt) == str(rcv_pkt):
+                match_index = ports.index(rcv_port)
+                received = True
+    verify_no_other_packets(test)
+
+    test.assertTrue(received == True, "Did not receive pkt on any of ports %r for device %d" % (ports, device_number))
+    return match_index
+
+def verify_each_packet_on_each_port(test, pkts=[], ports=[], device_number=0):
+    """
+    Check that each packet is received on corresponding port in the port list belonging to
+    the given device (default device_number is 0).
+
+    Also verifies that the packet is ot received on any other ports for this
+    device, and that no other packets are received on the device (unless --relax
+    is in effect).
+    """
+    pkt_cnt = 0
+    test.assertTrue(len(pkts) == len(ports), "packet list count does not match port list count")
+    for device, port in ptf_ports():
```

Comment:
is there a reason why you don't simply loop over the ports argument? This function is for a given device only anyway.
would this code work:
```
for port, pkt in zip(ports, pkts):
    // call to dp_poll
    test.assertTrue(rcv_pkt != None, ...)

verify no_other_packets(test, device_number)
```
I think the zip() call fits perfectly here.